tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-dashboard.html

```diff
@@ -277,8 +288,43 @@ <h3>No scalar data was found.</h3>
         return _smoothingWeight > 0;
       },
 
-      _shouldOpen(index) {
-        return index <= 2;
+      _getCategoryKey(category) {
+        return category.metadata.type ==
+            tf_categorization_utils.CategoryType.SEARCH_RESULTS ?
+                'search' : category.name;
+      },
+
+      _updatePaneOpened() {
+        const paneOpen = this._paneOpened;
+        const nextPaneOpen = {};
+
+        // Copy over the relevant values from the old pane state.
+        this._categories
+            .map(this._getCategoryKey)
+            .filter(name => paneOpen[name] != null)
+            .forEach(name => nextPaneOpen[name] = paneOpen[name]);
+
+        if (Object.keys(nextPaneOpen).length == 0) {
+          this._categories.slice(0, 2)
```

Comment:
Please make 2 a named constant or at least comment somewhere that it's a deliberate choice to open the first 2 panes by default, since in this context it's less self-documenting than before.